### DISCUSSION TITLE
Google Sign-In on Android [Ionic][Capacitor][Vue]

### DISCUSSION THREAD

**Discussion by bridgecode** [capawesome-team/capacitor-firebase#769] [Q&A] [answered]
Hello,

I successfully got the plugin working and everything seemed like it was okay until i tried to sign in. I keep getting an error that prevents the user from completing log in. I'm sure there's no issue with the plugin itself and likely my configuration. I was just curious if anyone else has encountered this issue and can maybe walk me through resolving this.

Any help would be greatly appreciated!

com.google.android.gms.common.api.ApiException: 10: 

![image](https://github.com/user-attachments/assets/f7d30f15-9ffb-4d6f-868f-7b62e37ffb84)

**Comment by robingenz** [comment, marked as answer]
`ApiException 10` says that you have specified the wrong or no SHA-1 key, see https://stackoverflow.com/a/49800546/6731412.

**Comment by bridgecode** [comment]
Hey @robingenz ,

Thanks for the super quick response and pointing me in the right direction! I'm new to status codes in Android, and it wasn’t immediately clear to me that the issue was with my SHA configuration, so your comment really helped.

Here’s what I did to resolve the issue, in-case anyone in the future has this issue:

- I deleted everything related to my debug and production keystores to start fresh.
- Recreated the debug keystore and updated the password.
- Recreated the production keystore with a new password.
- Updated Firebase with the new SHA-1 and SHA-256 fingerprints for both debug and production keys.
- Exported the updated google-services.json from Firebase.
- Replaced the google-services.json file in my project.
- Rebuilt everything from scratch, both debug and production APKs.

Some things to note:

1. I was using the debug .apk purely out of habit and didn’t realize that testing with the production .apk might have resolved the issue sooner.
2. Both debug and production builds should work if configured properly. If one isn’t working, try switching between them to isolate the issue.
3. If everything fails, starting from scratch by recreating the keystores, updating Firebase, and syncing the project is a solid way to fix it.

Again, thanks for the help, RG! This experience gave me a much better understanding of SHA configuration and keystore management.
